server/session.go

```diff
@@ -124,6 +124,7 @@ func (p *Plugin) addUserSession(userID, connID string, channel *model.Channel) (
 		if userID == botID {
 			if state.Call.Recording != nil && state.Call.Recording.StartAt == 0 {
 				state.Call.Recording.StartAt = time.Now().UnixMilli()
+				state.Call.Recording.BotConnID = connID
```

Comment:
I like the change, but I'm not sure I totally understand what the race condition was.

Reply:
@cpoile Apologies, let me try to explain it properly with a triggering sequence:

With a recording already ongoing:

1. Host stops recording
2. Recording state is saved as ended and websocket event sent out.
3. Client receives ws event and updates local state, recording has ended.
4. Host quickly starts a new recording before recording bot from previous recording leaves call.
6. Recording bot for previous recording job eventually leaves the call, improperly setting the current recording as ended.

Reply:
Were you able to reproduce this? If so, does the second recording start? (considering we're using the same channel state)

Reply:
Yes it's pretty easy to reproduce. Recording starts but the client gets the stop event right away and then the state is tainted so both client and server think there's no recording on-going while the job is actually running and recording until the call ends.

Reply:
Cool :)